application/src/main/java/org/togetherjava/tjbot/moderation/ModAuditLogWriter.java

```diff
@@ -0,0 +1,111 @@
+package org.togetherjava.tjbot.moderation;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.requests.restaction.MessageAction;
+import net.dv8tion.jda.api.utils.AttachmentOption;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.config.Config;
+
+import java.awt.*;
+import java.io.File;
+import java.nio.charset.StandardCharsets;
+import java.time.temporal.TemporalAccessor;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class that allows you to easily log an entry on the mod audit log channel. Thread-Safe.
+ */
+public enum ModAuditLogWriter {
+    ;
+    private static final Color EMBED_COLOR = Color.decode("#3788AC");
+
+    private static final Logger logger = LoggerFactory.getLogger(ModAuditLogWriter.class);
+
+    private static Pattern auditLogChannelNamePattern = null;
+
```

Comment:
This class must not be `static`, otherwise it can not be mocked for unit testing.

Very soon we will merge a PR that will make the `Config` a non-singleton. So you have to take the `Config` per constructor argument. This means that this class must also be non-static and use a proper constructor, otherwise it can not take the `Config` to determine the `auditLogChannelNamePattern`.

This will also bleed into `TagManageCommand` which must take an instance of `ModAuditLogWriter` per constructor parameter then.

Reply:
Not sure what am I supposed to do. should I make this a singleton, or have the use case instantiate a new object each time?

Reply:
I think I did it correctly.

Reply:
is the current state of `ModAuditLogWriter` good?